lib/stm32/h7/pwr.c

```diff
@@ -37,14 +30,31 @@
  */
 
 #include <libopencm3/stm32/pwr.h>
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/syscfg.h>
 
-/**@{*/
+
+void pwr_set_mode_ldo(void) {
+	const uint32_t ldo_mask = (PWR_CR3_SCUEN | PWR_CR3_LDOEN | PWR_CR3_BYPASS);
+	PWR_CR3 = (PWR_CR3 & ~ldo_mask) | (PWR_CR3_SCUEN | PWR_CR3_LDOEN);
+	while (!(PWR_CSR1 & PWR_CSR1_ACTVOSRDY));
+}
 
 void pwr_set_svos_scale(enum pwr_svos_scale scale)
 {
 	uint32_t pwr_cr1_reg = PWR_CR1;
-	pwr_cr1_reg = (pwr_cr1_reg & ~PWR_CR1_SVOS_MASK) | scale;
-	PWR_CR1 = pwr_cr1_reg;
+	pwr_cr1_reg = (pwr_cr1_reg & ~(PWR_CR1_SVOS_MASK  << PWR_CR1_SVOS_SHIFT));
+	PWR_CR1 = pwr_cr1_reg | scale;
 }
 
-/**@}*/
\ No newline at end of file
+void pwr_set_vos_scale(enum pwr_vos_scale scale) {
+	uint32_t d3cr_masked = PWR_D3CR & ~(PWR_D3CR_VOS_MASK << PWR_D3CR_VOS_SHIFT);
+	if (scale == PWR_VOS_SCALE_0) {
```

Comment:
Where does VOS_SCALE_0 even come from? there's no scale 0 in the docs.  Is this you attempting to set scale 1 if people use an unitialized variable?  I'd argue that either ignoring the request, errorring, blocking loop, or setting the default (scale 3) would all make more sense than hard setting to scale 1

Reply:
(are you actually checking for vos scal1 here for the overdrive? might be worth a comment)

Reply:
![image](https://user-images.githubusercontent.com/2933763/73961820-520f6600-48db-11ea-92b4-3fc8c1cdb084.png)

This is a value in the sheet, which is equivalent to VOS1 + boost/overdrive.

I've added a comment.